### ISSUE TITLE
Window is not defined

### ISSUE THREAD

**Issue by CodersGas** [deepakaggarwal7/react-social-login#191]
Hi! 
After updating to the latest version of react-social-login, I am getting window is not defined and its a package error.

Earlier I was using version 3.4.9 and it was working fine. In version 3.4.12, its crashing the website.

Node version - 14.15.4
Next js version - 10.0.4

![react-social-login](https://user-images.githubusercontent.com/42844733/109468837-3604d680-7a93-11eb-8b98-6b9dd0de0caa.png)

**Comment by gino8080** [comment]
same here

**Comment by simmbiote** [comment]
Same here. This is a breaking issue.

**Comment by CodersGas** [comment]
Yes. I think some dependency updation broke the package...

**Comment by CodersGas** [comment]
Yes. I think some dependency updation broke the package...

**Comment by BereniceQueiroz** [comment]
Olá por favor, conseguiram ajustar?
Se sim, poderiam informar como?

**Comment by CodersGas** [comment]
> Olá por favor, conseguiram ajustar?
> Se sim, poderiam informar como?

Sorry, but could you please type in english??

**Comment by BereniceQueiroz** [comment]
Hello please, managed to adjust?
If so, how?

**Comment by CodersGas** [comment]
> Hello please, managed to adjust?
> If so, how?

No. Not till now.

**Comment by deepakaggarwal7** [comment]
I am unable to replicate. Steps, I performed:

1. created a react-app
2. Installed latest react-social-login
3. followed the npm page

Facebook popup opened properly and I was able to login. Please ensure, you are using https and valid AppId.
If this doesn't help, I request to add more details/small video/not working gist etc. to help me fix.

**Comment by deepakaggarwal7** [comment]
Working sandbox https://codesandbox.io/s/react-social-login-h1tls?file=/src/App.js

**Comment by igorkurochka** [comment]
@deepakaggarwal7 I believe the problem is when we use the `react-social-login` plugin with `next` which uses SSR and forbids direct access to window object
try to create a simple next app, integrate plugin and then make `next build`
I have this error raised:

```
> Build error occurred
ReferenceError: window is not defined
    at Object.<anonymous> (/Users/bird/code/renterbuyer/frontbuild/node_modules/react-social-login/dist/social-login.js:1:299)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.FZEq (/Users/bird/code/renterbuyer/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:1749:18)
    at __webpack_require__ (/Users/bird/code/renterbuyer/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:29:31)
    at Object.qLLK (/Users/bird/code/renterbuyer/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:3184:36) {
  type: 'ReferenceError'
}
```

hope this helps

**Comment by igorkurochka** [comment]
the temporary solution for me for now will be to fix the version to "3.4.11" and to not update to the "3.4.12" where the problem is introduced

**Comment by deepakaggarwal7** [comment]
Благодаря @igorkurochka  ! that's helpful. If it's working 3.4.11 is same as 3.4.10 because the webpack step was missed. I can scan for code changes in last two versions and try with a nextjs app.

**Comment by deepakaggarwal7** [comment]
Can someone please check, if this line is breaking? Other than this, the last few changes do not have anything to do with window object (I've upgraded WebPack to v4, but I believe it should not emit breaking code)
```
    if (!navigator.onLine && this.props.onInternetFailure) {
      let shouldSkip = this.props.onInternetFailure()
      if (shouldSkip === false) {
        return
      }
    }
```

**Comment by CodersGas** [comment]
> @deepakaggarwal7 I believe the problem is when we use the `react-social-login` plugin with `next` which uses SSR and forbids direct access to window object
> try to create a simple next app, integrate plugin and then make `next build`
> I have this error raised:
> 
> ```
> > Build error occurred
> ReferenceError: window is not defined
>     at Object.<anonymous> (/Users/bird/code/renterbuyer/frontbuild/node_modules/react-social-login/dist/social-login.js:1:299)
>     at Module._compile (internal/modules/cjs/loader.js:1137:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
>     at Module.load (internal/modules/cjs/loader.js:985:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:878:14)
>     at Module.require (internal/modules/cjs/loader.js:1025:19)
>     at require (internal/modules/cjs/helpers.js:72:18)
>     at Object.FZEq (/code/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:1749:18)
>     at __webpack_require__ (/code/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:29:31)
>     at Object.qLLK (/code/frontbuild/.next/server/static/8URcg8nyWN3MR28oA-cRS/pages/signup.js:3184:36) {
>   type: 'ReferenceError'
> }
> ```
> 
> hope this helps

Yes, this is the true error form. Sorry I didn't posted it properly.

**Comment by igorkurochka** [comment]
> Благодаря @igorkurochka ! that's helpful. If it's working 3.4.11 is same as 3.4.10 because the webpack step was missed. I can scan for code changes in last two versions and try with a nextjs app.

I can confirm it's working on 3.4.11

**Comment by ankit-vinove** [comment]
no it is not working with 3.4.11 i had tried just now

**Comment by igorkurochka** [comment]
> no it is not working with 3.4.11 i had tried just now

Oh, that's weird, it definitely works for me...

**Comment by ankit-vinove** [comment]
I don't know what magic happen but again i had tried and now it is running fine😂 . 
thanks @igorkurochka  for your help!!!!

**Comment by deepakaggarwal7** [comment]
Fixed in 3.4.13.
